finality-aleph/src/network/service.rs

```diff
@@ -25,8 +26,10 @@ pub struct Service<N: Network, D: Data> {
     messages_from_user: mpsc::UnboundedReceiver<(D, DataCommand)>,
     messages_for_user: mpsc::UnboundedSender<D>,
     commands_from_manager: mpsc::UnboundedReceiver<ConnectionCommand>,
-    connected_peers: HashSet<PeerId>,
-    peer_senders: HashMap<PeerId, mpsc::Sender<(D, Protocol)>>,
+    generic_connected_peers: HashSet<PeerId>,
+    validator_connected_peers: HashSet<PeerId>,
+    generic_peer_senders: HashMap<PeerId, mpsc::Sender<(D, Protocol)>>,
+    validator_peer_senders: HashMap<PeerId, mpsc::Sender<(D, Protocol)>>,
```

Comment:
Since you now start them per protocol, you don't need to pass the protocol to them. This requires some changes to `peer_sender`, but you can remove `Protocol` from the pair here.

Reply:
Done